src/vs/workbench/parts/extensions/common/extensionQuery.ts

```diff
@@ -21,11 +21,25 @@ export class Query {
 			'ext': ['']
 		};
 
+		let containsSub = (haystack: string) => (needle: string) => haystack.indexOf(needle) !== -1;
+		let queryContains = containsSub(query);
+		let hasSort = subcommands.sort.some(subcommand => queryContains(`@sort:${subcommand}`));
+		let hasCategory = subcommands.category.some(subcommand => queryContains(`@category:${subcommand}`));
+
 		return flatten(
-			commands.map(command =>
-				subcommands[command]
-					? subcommands[command].map(subcommand => `@${command}:${subcommand}${subcommand === '' ? '' : ' '}`)
-					: [`@${command} `]));
+			commands.map(command => {
+				let commandContains = containsSub(command);
+				if (hasSort && commandContains('sort') || hasCategory && commandContains('category')) {
```

Comment:
Dont we need an equals check here instead of contains?

Reply:
Could you clarify?